src/lib/stateMappings/singleIndex.ts

```diff
@@ -0,0 +1,21 @@
+import { StateMapping, IndexUiState } from '../../types';
+
+function getIndexStateWithoutConfigure(uiState: IndexUiState): IndexUiState {
+  const { configure, ...trackedUiState } = uiState;
+  return trackedUiState;
+}
+
+export default function singleIndexStateMapping(
+  rootIndexName: string
```

Comment:
Practically, this is not always the root index name but could be any index name. We can call the parameter `indexName`.